code_annotations/extensions/base.py

```diff
@@ -19,14 +26,219 @@ def __init__(self, config, echo):
         self.config = config
         self.ECHO = echo
 
-    def validate(self, file_handle):
+        annotation_tokens = config['annotations']
+
+        # Allow sub-classes to configure themselves with the configured annotations
+        for annotation_or_group in annotation_tokens:
+            if isinstance(annotation_tokens[annotation_or_group], six.string_types):
+                self._add_annotation_token(annotation_tokens[annotation_or_group])
+            elif isinstance(annotation_tokens[annotation_or_group], (list, tuple)):
+                self._add_annotation_group(annotation_tokens[annotation_or_group])
+            elif isinstance(annotation_tokens[annotation_or_group], dict):
+                self._add_annotation_token(next(iter(annotation_tokens[annotation_or_group])))
+            else:
+                raise TypeError('{} is an unknown type. Annotations must be strings, list/tuples, or dicts.'.format(
+                    annotation_tokens[annotation_or_group])
+                )
+
+    def validate(self, file_handle):  # pragma: no cover
```

Comment:
And shouldn't `validate`, `search`, and `_add_annotation_*` all be decorated as `@abstractmethod`?